pyqt-apps/siriushla/as_di_bpms/main.py

```diff
@@ -45,6 +48,54 @@ def setupui(self):
         self.layoutv.addWidget(grpbx)
         self.layoutv.addSpacing(20)
         self.layoutv.addStretch()
+        if self.bpm[:2] == 'SI':
+            if self.bpm[5:9] == 'SBFE' or self.bpm[5:9] == 'SPFE' or self.bpm[5:9] == 'SAFE':
+                self._amcFRU(4)
+
+            elif self.bpm[5:9] == 'BCFE':
+                self._amcFRU(5)
+
+            elif self.bpm[5:8] == 'SA:' or self.bpm[5:8] == 'SB:' or self.bpm[5:8] == 'SP:':
+                self._amcFRU(6)
+
+            elif self.bpm[5:7] == 'M1' or self.bpm[5:7] == 'M2':
+                self._amcFRU(7)
+
+            elif self.bpm[5:7] == 'C1':
+                self._amcFRU(8)
+
+            elif self.bpm[5:7] == 'C2' or self.bpm[5:16] == 'C3:DI-BPM-1':
+                self._amcFRU(9)
+
+            elif self.bpm[5:7] == 'C4' or self.bpm[5:16] == 'C3:DI-BPM-2':
+                self._amcFRU(10)
+
+        elif self.bpm[:2] == 'BO':
+            if int(self.bpm[3:5]) in afcs_pvs.BO_N or int(self.bpm[3:5]) - 1 in afcs_pvs.BO_N:
+                self._amcFRU(11)
+
+            elif int(self.bpm[3:5]) - 2 in afcs_pvs.BO_N:
+                self._amcFRU(12)
+
+        elif self.bpm[:2] == 'TB':
+            if self.bpm[3:5] == '01':
+                self._amcFRU(6)
+
+            elif self.bpm[3:5] == '02':
+                self._amcFRU(7)
+
+            elif self.bpm[3:5] == '03' or self.bpm[3:5] == '04':
+                self._amcFRU(8)
+
+        elif self.bpm[:2] == 'TS':
+            if self.bpm[3:5] == '01' or self.bpm[3:5] == '02':
+                self._amcFRU(9)
+
+            elif self.bpm[3:5] == '03' or self.bpm[3:14] == '04:DI-BPM-1':
+                self._amcFRU(10)
+
+            elif self.bpm[3:14] == '04:DI-BPM-2':
+                self._amcFRU(11)
```

Comment:
you can use `SiriusPVName` class here for string parsing:
section -> `SiriusPVName.sec`
subsection -> `SiriusPVName.sub`
discipline -> `SiriusPVName.dis`
device -> `SiriusPVName.dev`
index -> `SiriusPVName.idx`

`self.bpm` is already an instance of `SiriusPVName`, so, for example, you can replace:
- `self.bpm[:2] ` by `self.bpm.sec
- for TB, BO or TS `self.bpm[3:5]` by `self.bpm.sub
- for SI, `self.bpm[5:*]` by self.bpm.sub.endswith(...)